spec/models/map_spec.rb

```diff
@@ -553,6 +553,24 @@ def become_viewer(user)
 
         @map.can_add_layer(@user).should eq false
       end
+
+      it 'should take shared_empty_dataset into account' do
+        emptyDatasetName = Cartodb.config[:shared_empty_dataset_name]
+        @map = Map.create(user_id: @user.id, table_id: @table.id)
+        @map.reload
+
+        @layer = Layer.create(kind: 'carto', options: { :table_name => emptyDatasetName })
+        @map.add_layer(@layer)
+        @map.reload
+
+        for lyr in 1..@user.max_layers do
+          @map.can_add_layer(@user).should eq true
+          @map.add_layer(Layer.new(kind: 'carto'))
+          @map.reload
+        end
+
+        @map.can_add_layer(@user).should eq false
```

Comment:
Be sure to end the spec destroying any objects you created in it. Here, you'd be missing

```ruby
@map.destroy
```

Reply:
already done by line 505